src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeoAuthProvider.scala

```diff
@@ -25,45 +24,31 @@ object NotebookClusterActions {
   case object SyncDataToCluster extends NotebookClusterAction
   case object DeleteCluster extends NotebookClusterAction
   val allActions = Seq(GetClusterStatus, ConnectToCluster, SyncDataToCluster, DeleteCluster)
-
 }
 
 abstract class LeoAuthProvider(authConfig: Config, serviceAccountProvider: ServiceAccountProvider) {
 
   /**
-    * @param userEmail The user in question
+    * @param userInfo The user in question
     * @param action The project-level action (above) the user is requesting
     * @param googleProject The Google project to check in
     * @return If the given user has permissions in this project to perform the specified action.
     */
-  def hasProjectPermission(userEmail: WorkbenchEmail, action: ProjectActions.ProjectAction, googleProject: GoogleProject)(implicit executionContext: ExecutionContext): Future[Boolean]
-
-  /**
-    * When listing clusters, Leo will perform a GROUP BY on google projects and call this function once per google project.
-    * If you have an implementation such that users, even in some cases, can see all clusters in a google project, overriding
-    * this function may lead to significant performance improvements.
-    * For any projects where this function call returns Future.successful(false), Leo will then call hasNotebookClusterPermission
-    * for every cluster in that project, passing in action = GetClusterStatus.
-    *
-    * @param userEmail The user in question
-    * @param googleProject A Google project
-    * @return If the given user can see all clusters in this project
-    */
-  def canSeeAllClustersInProject(userEmail: WorkbenchEmail, googleProject: GoogleProject)(implicit executionContext: ExecutionContext): Future[Boolean] = {
-    Future.successful(false)
-  }
+  def hasProjectPermission(userInfo: UserInfo, action: ProjectActions.ProjectAction, googleProject: GoogleProject)(implicit executionContext: ExecutionContext): Future[Boolean]
```

Comment:
We now pass `hasProjectPermission` and `hasNotebookClusterPermission` a `UserInfo` instead of a `WorkbenchEmail`. Leo never needs to call these offline, so we always have a `UserInfo`. `UserInfo` contains an email+token, so the provider can decide which to use.

Reply:
OK. We should be aware that we've flip-flopped whether to use `UserInfo` or `UserEmail` in SA-provider signatures a bunch of times now. It's not a problem _yet_ but at some point we should start considering this interface immutable so not screw with people adding their own providers on the classpath.

Reply:
Yes, agreed. As far as I know no one has custom implementations of these providers yet (people are using `DefaultServiceAccountProvider` and `WhitelistAuthProvider` but those are still managed by us.